selfupdate/selfupdate.go

```diff
@@ -0,0 +1,179 @@
+package selfupdate
+
+import (
+	"context"
+	"fmt"
+	"github.com/Graylog2/graylog-project-cli/ask"
+	"github.com/Graylog2/graylog-project-cli/logger"
+	"github.com/fatih/color"
+	"github.com/google/go-github/v60/github"
+	"github.com/hashicorp/go-version"
+	"github.com/mattn/go-isatty"
+	"github.com/samber/lo"
+	"github.com/schollz/progressbar/v3"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+)
+
+const (
+	repoOwner = "Graylog2"
+	repoName  = "graylog-project-cli"
+)
+
+func SelfUpdate(runningVersion *version.Version, requestedVersion string, force bool, interactive bool) error {
+	// Find the current binary first
+	binPath, err := exec.LookPath(os.Args[0])
+	if err != nil {
+		return fmt.Errorf("couldn't find binary path for %q: %w", os.Args[0], err)
+	}
+
+	binFileInfo, err := os.Stat(binPath)
+	if err != nil {
+		return fmt.Errorf("couldn't get file info for current binary %q: %w", binPath, err)
+	}
```

Comment:
We have to consider symlinks. Probably something like this should work.

```suggestion
	binFileInfo, err := os.Lstat(binPath)
	if err != nil {
		return fmt.Errorf("couldn't get file info for current binary %q: %w", binPath, err)
	}

	if binFileInfo.Mode() & os.ModeSymlink == os.ModeSymlink {
		linkTarget, err := os.Readlink(binPath)
		if err != nil {
			return fmt.Errorf("couldn't resolve symlink for current binary %q: %w", binPath, err)
		}

		binPath, err = filepath.Abs(linkTarget)
		if err != nil {
			return fmt.Errorf("couldn't resolve absolute path for current binary %q: %w", linkTarget, err)
		}
	}

```